demo/js/animated-dialog.js

```diff
@@ -0,0 +1,44 @@
+const { createFocusTrap } = require('../../dist/focus-trap');
+
+const container = document.getElementById('animated-dialog');
+
+const focusTrap = createFocusTrap('#animated-dialog', {
+  // Called before focus is sent
+  onActivate: () => container.classList.add('is-active'),
+  // There is a delay between when the class is applied
+  // and when the element is focusable
+  checkCanFocusTrap: (trapContainers) => {
+    const results = trapContainers.map(
+      (trapContainer) =>
+        new Promise((resolve) => {
+          const interval = setInterval(() => {
+            if (getComputedStyle(trapContainer).visibility !== 'hidden') {
+              resolve();
+              clearInterval(interval);
+            }
+          }, 5);
+        })
+    );
+    // Return a promise that resolves when all the trap containers are able to receive focus
+    return Promise.all(results).then(
+      () => 'The value that you return here will be sent to onPostActivate'
```

Comment:
I'd prefer not to support this return value until there's a clear use case for it.